Would you mind fixing up the failing test case?

```
--- FAIL: TestList_Sort (0.00s)
    list_test.go:97: 
        	Error Trace:	list_test.go:97
        	Error:      	Not equal: 
        	            	expected: "two"
        	            	actual  : "one"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-two
        	            	+one
        	Test:       	TestList_Sort
    list_test.go:99: 
        	Error Trace:	list_test.go:99
        	Error:      	Not equal: 
        	            	expected: "one"
        	            	actual  : "two"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-one
        	            	+two
        	Test:       	TestList_Sort
FAIL
```

--- a/cmd/helm/list.go
+++ b/cmd/helm/list.go
@@ -87,7 +87,7 @@ func newListCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 	f.BoolVarP(&client.Short, "short", "q", false, "output short (quiet) listing format")
 	f.BoolVarP(&client.ByDate, "date", "d", false, "sort by release date")
 	f.BoolVarP(&client.SortReverse, "reverse", "r", false, "reverse the sort order")
-	f.BoolVarP(&client.All, "all", "a", false, "show all releases, not just the ones marked deployed")
+	f.BoolVarP(&client.All, "all", "a", false, "show all releases, not just the ones marked deployed or failed")
 	f.BoolVar(&client.Uninstalled, "uninstalled", false, "show uninstalled releases")
 	f.BoolVar(&client.Superseded, "superseded", false, "show superseded releases")
 	f.BoolVar(&client.Uninstalling, "uninstalling", false, "show releases that are currently being uninstalled")

--- a/pkg/action/list.go
+++ b/pkg/action/list.go
@@ -87,10 +87,8 @@ const ListAll = ListDeployed | ListUninstalled | ListUninstalling | ListPendingI
 type Sorter uint
 
 const (
-	// ByNameAsc sorts by ascending lexicographic order
-	ByNameAsc Sorter = iota
 	// ByNameDesc sorts by descending lexicographic order
-	ByNameDesc
+	ByNameDesc Sorter = iota + 1
 	// ByDateAsc sorts by ascending dates (oldest updated release first)
 	ByDateAsc
 	// ByDateDesc sorts by descending dates (latest updated release first)

--- a/pkg/action/list.go
+++ b/pkg/action/list.go
@@ -87,14 +87,14 @@ const ListAll = ListDeployed | ListUninstalled | ListUninstalling | ListPendingI
 type Sorter uint
 
 const (
-	// ByDateAsc sorts by ascending dates (oldest updated release first)
-	ByDateAsc Sorter = iota
-	// ByDateDesc sorts by descending dates (latest updated release first)
-	ByDateDesc
 	// ByNameAsc sorts by ascending lexicographic order
-	ByNameAsc
+	ByNameAsc Sorter = iota
 	// ByNameDesc sorts by descending lexicographic order
 	ByNameDesc
+	// ByDateAsc sorts by ascending dates (oldest updated release first)
+	ByDateAsc
+	// ByDateDesc sorts by descending dates (latest updated release first)
+	ByDateDesc
 )
 
 // List is the action for listing releases.
@@ -197,7 +197,6 @@ func (l *List) Run() ([]*release.Release, error) {
 
 // sort is an in-place sort where order is based on the value of a.Sort
 func (l *List) sort(rels []*release.Release) {
-	l.Sort = ByNameAsc
 	if l.SortReverse {
 		l.Sort = ByNameDesc
 	}

--- a/pkg/action/list_test.go
+++ b/pkg/action/list_test.go
@@ -103,13 +103,10 @@ func TestList_Limit(t *testing.T) {
 	is := assert.New(t)
 	lister := newListFixture(t)
 	lister.Limit = 2
-	// Sort because otherwise there is no guaranteed order
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 	list, err := lister.Run()
 	is.NoError(err)
 	is.Len(list, 2)
-
 	// Lex order means one, three, two
 	is.Equal("one", list[0].Name)
 	is.Equal("three", list[1].Name)
@@ -119,8 +116,6 @@ func TestList_BigLimit(t *testing.T) {
 	is := assert.New(t)
 	lister := newListFixture(t)
 	lister.Limit = 20
-	// Sort because otherwise there is no guaranteed order
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 	list, err := lister.Run()
 	is.NoError(err)
@@ -137,8 +132,6 @@ func TestList_LimitOffset(t *testing.T) {
 	lister := newListFixture(t)
 	lister.Limit = 2
 	lister.Offset = 1
-	// Sort because otherwise there is no guaranteed order
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 	list, err := lister.Run()
 	is.NoError(err)
@@ -154,8 +147,6 @@ func TestList_LimitOffsetOutOfBounds(t *testing.T) {
 	lister := newListFixture(t)
 	lister.Limit = 2
 	lister.Offset = 3 // Last item is index 2
-	// Sort because otherwise there is no guaranteed order
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 	list, err := lister.Run()
 	is.NoError(err)
@@ -170,8 +161,6 @@ func TestList_LimitOffsetOutOfBounds(t *testing.T) {
 func TestList_StateMask(t *testing.T) {
 	is := assert.New(t)
 	lister := newListFixture(t)
-	// Sort because otherwise there is no guaranteed order
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 	one, err := lister.cfg.Releases.Get("one", 1)
 	is.NoError(err)
@@ -200,7 +189,6 @@ func TestList_Filter(t *testing.T) {
 	is := assert.New(t)
 	lister := newListFixture(t)
 	lister.Filter = "th."
-	lister.Sort = ByNameAsc
 	makeMeSomeReleases(lister.cfg.Releases, t)
 
 	res, err := lister.Run()

